ts/Core/Axis/Axis.ts

```diff
@@ -2976,35 +2976,35 @@ class Axis {
             }
 
             if (autoRotation) {
+                let rotStep,
+                    score;
 
-                // Loop over the given autoRotation options, and determine
-                // which gives the best score. The best score is that with
-                // the lowest number of steps and a rotation closest
-                // to horizontal.
-                autoRotation.forEach(function (rot: number): void {
-                    let score;
 
+                // Loop over the given autoRotation options, and determine which
+                // gives the best score. The best score is that with the lowest
+                // number of steps and a rotation closest to horizontal.
+                for (const rot of autoRotation) {
                     if (
                         rot === rotationOption ||
                         (rot && rot >= -90 && rot <= 90)
                     ) { // #3891
 
-                        step = getStep(
+                        rotStep = step || getStep(
                             Math.abs(labelMetrics.h / Math.sin(deg2rad * rot))
```

Comment:
I think the main issue is that we use line height to measure the size of horizontal ticks. We should have some line width measurement in place or otherwise it becomes always an unreliable guessing game.

Reply:
I think we do... We don't start rotating in the first place unless the labels are actually overlapping.